tests/lib/rules/no-useless-assignment.js

```diff
@@ -353,7 +369,39 @@ ruleTester.run("no-useless-assignment", rule, {
         } = obj;
         console.log(a, b);`,
         `let { a, b: {c = a} = {} } = obj;
-        console.log(c);`
+        console.log(c);`,
+        {
+            code: `/*eslint test/jsx:1*/
+                function App() {
+                    const A = "";
+                    return <A/>;
+                }
+            `,
+            languageOptions: {
+                parserOptions: {
+                    ecmaFeatures: {
+                        jsx: true
+                    }
+                }
+            }
+        },
+        {
+            code: `/*eslint test/jsx:1*/
+                function App() {
+                    let A = "";
+                    foo(A);
+                    a = "A";
+                    return <A/>;
+                }
+            `,
+            languageOptions: {
+                parserOptions: {
+                    ecmaFeatures: {
+                        jsx: true
+                    }
+                }
+            }
+        }
```

Comment:
```suggestion
        },
        {
            code: `/*eslint test/jsx:1*/
                function App() {
                    let A = "a";
                    A = "b";
                    A = "c";
                    foo(A);
                    return <A/>;
                }
            `,
            languageOptions: {
                parserOptions: {
                    ecmaFeatures: {
                        jsx: true
                    }
                }
            }
        }
```

One more test to confirm that these variables are entirely ignored.